feature/crosswalk

teamagam1 opened the pull request.

barmako committed:
Backoff mechanism implementations.
The integration isn't complete yet. as IntentService is incapable of rescheduling polling jobs in the future using a handler (since its thread is closing after its handling method is done).
A refactor needs to be made to change the message polling service into a ThreadHandler.

teamagam1 committed:
basic cross-walk works woth 32-bit lib.

CesiumMapView extends XWalkView.

- dynamic view inflation (not via xml)
- bridge with JS does not work yet.

teamagam1 committed:
works for all devices using maven repo

teamagam1 committed:
changed gg_map_view name to fragment container

teamagam1 committed:
cross walk works with xml in layout file

teamagam1 committed:
Java script interface works

barmako committed:
Fully working exponential back-off message polling.

Message polling is now done in a separate  dedicated thread instead of using an intent service, hence the service java file has been deleted and its references across the application and the manifest file as-well.

Polling starts and ends just as before, with main activity active status (between onResume and onPause).

The repeatable action running that depends on some sort of backoff mechanism was written in its own class RepeatedBackoffTaskRunner.
It has its own implementation that instantiate it in a way that it uses a dedicated  thread (it is possible to make others TaskRunners that would use the same thread)
It uses an exponential backoff strategy and does the polling action via the IMessagePoller, made earlier (way-way earlier....)
The packages hierarchy changed a bit to fit the new state where no service is used for the message polling.

Auto-indentation across all "touched" files

teamagam1 committed:
with console log for error console messages cesium

teamagam1 committed:
using local module and xml view

teamagam1 committed:
xwalk_core_library

teamagam1 committed:
few fixes and onLoadFinshed moved inside XWalkResourceClient.

teamagam1 committed:
fixed buildToolsVersion of xwalk

teamagam1 committed:
XWalk lifecycle fixed.

teamagam1 committed:
disabling fullscreen button with destroy.

review-ninja commented:
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/67" target="_blank"><img src="https://app.review.ninja/52010471/pull/67/badge" alt="ReviewNinja"/>

teamagam1 committed:
hidden credit

teamagam1 committed:
fullscreenButton created to be destroyed

teamagam1 committed:
onCesiumReady fix

BarAgam commented on app/src/main/assets/GimelGimel/GGBootstrap.js:
```diff
@@ -23,8 +23,12 @@ GG.viewer = new Cesium.Viewer('cesiumContainer', {
+    creditContainer: "creditContainer"
 });
 
+//cross-view hack.
```
Make a detailed comment !fix

teamagam1 replied:
!fixed

BarAgam commented on app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/GGMapView.java:
```diff
@@ -37,5 +37,15 @@
     void restoreViewState(Bundle inState);
+
+    /**
+     * pauses the view. needed for pausing javascript timers.
```
Delete !fix

teamagam1 replied:
!fixed

BarAgam commented on app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java:
```diff
@@ -62,20 +63,12 @@
-    }
 
     @Override
     public void onDetachedFromWindow() {
```
move thid method downwards

teamagam1 committed:
CR fixes

BarAgam commented on app/src/main/res/layout/fragment_cesium.xml:
```diff
@@ -8,32 +8,33 @@
         android:layout_width="match_parent"
-        android:layout_height="match_parent"/>
+        android:layout_height="match_parent"
+        android:hardwareAccelerated="true"/>
```
Auto-indent this file !Fix

barmako committed:
Forced the creation of polling task via the application's context (to avoid memory leaks)
Used GGApplication's SecuredPreferenceUtil instead of creating my own PreferenceUtil

barmako committed:
Forced the creation of polling task via the application's context (to avoid memory leaks)
Used GGApplication's SecuredPreferenceUtil instead of creating my own PreferenceUtil.

renamed poller dir to polling.

teamagam1 committed:
Merge remote-tracking branch 'origin/develop' into feature/crosswalk
